File: AndroidUsingJava/DesignPatterns/newMVP2/app/src/main/java/com/example/newmvp/db/ProductDAO.java
package com.example.newmvp.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.newmvp.model.Product;

import java.util.List;

@Dao
public interface ProductDAO {
    @Query("SELECT * FROM products_table")
    LiveData<List<Product>> getFavProducts();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertProduct(Product product);

    @Delete
    void deleteProduct(Product product);
}

File: AndroidUsingKotlin/Session_7/MVVM/app/src/main/java/com/example/mvvm/DB/ProductDao.kt
package com.example.mvvm.DB

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.mvvm.Product

@Dao
interface ProductDao {

    @Query("SELECT * FROM Products")
    suspend fun getAllProducts(): List<Product>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertAll(products: List<Product>) : List<Long>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(product:Product) : Long

    @Delete
    suspend fun delete(product: Product): Int
}